### ISSUE TITLE
Map keys order it's not being persistent through different spruce merges

### ISSUE THREAD

**Issue by qu1queee** [geofffranks/spruce#184]
There is a strange behaviour when merging this:

**File A.yml**
```yaml
properties:
  cc:
    quota_definitions:
      q256MB:
        non_basic_services_allowed: true
      q2GB:
        non_basic_services_allowed: true
      q4GB:
        non_basic_services_allowed: true
```

**File B.yml**
```yaml
properties:
  cc:
    quota_definitions:
      q2GB:
        non_basic_services_allowed: false
      q4GB:
        non_basic_services_allowed: false
```

When running several times the `spruce merge A.yml B.yml`. The `properties.cc.quota_definitions` map keys are having not a persistent order.

@geofffranks pointed out that this is a bug in the go-yaml library, as stated here: https://github.com/go-yaml/yaml/issues/194

And there is a proposed fix here: https://github.com/go-yaml/yaml/pull/195

In conclusion, the yaml marshaller is having problems when the map keys start with the same digit, where the logic skips them and calculates the order based on the rest of the digits. 

In my case, modifying the order of an important property like the one of the `properties.cc.quota_defintions` after every merge, will cause a restart of the agents inside the BOSH jobs that uses this property(unwanted behaviour)
